electron-app/src/server/submission/file-submission/file-submission.service.ts

```diff
@@ -26,6 +32,11 @@ export class FileSubmissionService {
 
     const title = path.parse(file.originalname).name;
     const locations = await this.fileRepository.insertFile(submission._id, file, data.path);
+    const im: ImageManipulator = await this.imageManipulationPool.getImageManipulator(
```

Comment:
concern: does this throw exception if something like a gif or text file is provided?

Reply:
Tweaked :) Well spotted!